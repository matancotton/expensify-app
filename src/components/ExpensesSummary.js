import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-toal';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{props.expenseCount}</span> {expenseWord} totalling <span>{numeral(props.expensesTotal / 100).format('$0,0.00')}</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => ({
    expenseCount: selectExpenses(state.expenses,state.filters).length,
    expensesTotal: getExpensesTotal(selectExpenses(state.expenses,state.filters))
})

export default connect(mapStateToProps)(ExpensesSummary);