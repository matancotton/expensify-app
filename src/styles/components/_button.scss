.button {
    background: $light-blue;
    color: white;
    display: inline-block;
    border: none;
    padding: $s-size;
    font: 300 $font-size-large;
    text-decoration: none;
    line-height: 1;
}

.button--secondary {
    background: #888888;
}

.button--link {
    background: none;
    border: none;
    color: white;
    font-size: 1.8rem;
    font-weight: 300;
}