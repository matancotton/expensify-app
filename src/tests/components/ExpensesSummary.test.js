import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 1 expense totalling $94.34', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpensesSummary with 2 expense totalling $94.34', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
})