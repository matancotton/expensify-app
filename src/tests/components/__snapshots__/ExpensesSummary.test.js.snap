// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense totalling $94.34 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with 2 expense totalling $94.34 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling 
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;
